src/Agent.Worker/Build/ArtifactCommandExtension.cs

```diff
@@ -30,6 +33,10 @@ public void ProcessCommand(IExecutionContext context, Command command)
             {
                 ProcessArtifactUploadCommand(context, command.Properties, command.Data);
             }
+            else if (string.Equals(command.Event, WellKnownArtifactCommand.Download, StringComparison.OrdinalIgnoreCase))
```

Comment:
A download command should not be added. Download should be exposed as a task only.

A command makes sense for upload, not download.